src/Swarm/TUI/Model.hs

```diff
@@ -308,15 +321,19 @@ populateInventoryList :: MonadState UIState m => Maybe Robot -> m ()
 populateInventoryList Nothing = uiInventory .= Nothing
 populateInventoryList (Just r) = do
   mList <- preuse (uiInventory . _Just . _2)
-  let mkInvEntry (n, e) = InventoryEntry n e
-      itemList label =
-        (\case [] -> []; xs -> Separator label : xs)
-          . map mkInvEntry
-          . sortOn (view entityName . snd)
-          . elems
-      items =
-        (r ^. robotInventory . to (itemList "Inventory"))
-          ++ (r ^. installedDevices . to (itemList "Installed devices"))
+  search <- use (uiSearch)
+  let mkInvEntry (n,e) = InventoryEntry n e
+      itemList label
+        = (\case { [] -> []; xs -> Separator label : xs })
+        . map mkInvEntry
+        . filter (T.isPrefixOf search . view entityName . snd)
+        . sortOn (view entityName . snd)
+        . elems
+      allItems = (r ^. robotInventory . to (itemList "Inventory"))
+              ++ (r ^. installedDevices . to (itemList "Installed devices"))
+      items
+        | T.null search = allItems
+        | otherwise     = (Separator $ "Search: " <> search) : allItems
```

Comment:
when searching, i think it would be nice to adjust the currently selected item to skip the separator, otherwise when starting a search, the `Installed devices` separator is selected.

Reply:
Yeah, this is a bit weird.